### ISSUE TITLE
Edited page source is not saved when switching back between WYSIWYG and Source

### ISSUE THREAD

**Issue by clelange** [indico/indico#4616]
**Describe the bug**
Saving a page is disabled unless one has performed changes using the WYSIWYG editor. If one edits a page using the "Source" button, the "Save" button is disabled until one switches back to the WYSIWYG editor. Changes are correctly propagated to the WYSIWYG editor. However, if one switches back to the Source again, one can save the page, but any change performed directly in the source is lost.

**To Reproduce**
1. Go to editing a page (menu item)
2. Click on "Source" in the editor
3. Perform changes to the source
4. Go back to the editor by clicking on "Source" again
5. See that changes are propagated from the source and that the Save button is enabled
6. Go back to the source
7. Make further changes in the source
8. Click save.

**Expected behavior**
At step 7, the save button should be disabled again or the changes should be propagated to the page, but not lost.

**Comment by ThiefMaster** [comment]
AFAIK ckeditor provides no way to act whenever the user switches to the source view, so this is most likely not possible.

**Comment by clelange** [comment]
What about saving the changes made in the source without the need to switch back to WYSIWYG instead of ignoring them?

**Comment by ThiefMaster** [comment]
IIRC not possible either. The editor does not populate its hidden field until you leave source mode.

**Comment by clelange** [comment]
In this case, it would be great if there was a warning that the source mode doesn't save changes, e.g. when clicking on the source button. It's kind of frustrating if everything is lost...

**Comment by ThiefMaster** [comment]
I agree but I'm not sure how many people even use it... showing the warning to everyone might be more confusing than helpful to people who don't even know what HTML or "source" is ;)

That said, I wonder if we could do something hacky to detect from outside the editor itself whether the source view is active and use that to disable the submit button...
